department-of-veterans-affairs/caseflow
# Fix case search access control

Resolves #6933. Makes it possible for non-queue users to access case details pages.

Additionally, this PR removes the concept of a "queue user". If somebody who would previously have been prevented from visiting `/queue` navigates to `/queue` they will now see an empty page with a case search bar instead of being redirected to `/unauthorized`. This potentially allows more people to access case search than were previously able to, but I think that group of people is small and if we can determine who they are we should explicitly limit their access to which cases they can search for (as we do with VSO employees).

Below is the list of people we were explicitly allowing access to `/queue` and, by extension, case search:
* System admins
* Organizational queue users
* VSO employees (explicitly prevented from seeing cases that they do not have BGS permissions for)
* Attorneys
* Judges
* Co-located administrative staff
* Anyone with the `case_search_home_page` function (no access to queue before)

## Test plan
* Log in as `CASE_SEARCHER_ONLY`
* Search for a Veteran's case (e.g. 239120550)
* Click on the link in the results page
* See the page successfully load
